From 17ce45639ac428228be390da0b065b6009f2b73e Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Wed, 11 Aug 2021 21:45:15 +0530
Subject: [PATCH] [REVERTME] Revert patch for allowing PHONY target

 commit 6587beddfa79ee06db08625a3673f6b97d7a68bd
 Author: Dan Willemsen <dwillemsen@google.com>
 Date:   Sat Apr 18 20:25:59 2020 -0700

Turn on ninja phony outputs

This replaces the _kati_always_build_ hack with a ninja feature so that
ninja can actually understand what's happening. This means that we can
turn on some more options and checks around expected output filenames:

* Remove the output file(s) before the command executes
* Error if the output file(s) don't exist after the execution
* Error if the output is a directory

They're turned on immediately during the soong bootstrap runs, as those
run a limited number of rules. The main ninja execution does not remove
the output files, and prints warnings instead of errors for the others.
I'll turn them on as we understand how often those warnings are seen.

Test: m (check for new warnings)
Test: treehugger (check for new warnings)
 Change-Id: I7f5c1a361dd1498eb54a2c07a918f3b0aa086e4c

Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Change-Id: I48bdf330cfc57e410c0ed0302e0d689fd0b642b2
---
 ui/build/build.go | 1 +
 ui/build/kati.go  | 2 +-
 ui/build/ninja.go | 2 +-
 ui/build/soong.go | 8 ++++----
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/ui/build/build.go b/ui/build/build.go
index 8f050d9be..b49f43743 100644
--- a/ui/build/build.go
+++ b/ui/build/build.go
@@ -60,6 +60,7 @@ builddir = {{.OutDir}}
 {{end -}}
 pool highmem_pool
  depth = {{.HighmemParallel}}
+build _kati_always_build_: phony
 {{if and (not .SkipKatiNinja) .HasKatiSuffix}}subninja {{.KatiBuildNinjaFile}}
 subninja {{.KatiPackageNinjaFile}}
 {{end -}}
diff --git a/ui/build/kati.go b/ui/build/kati.go
index fa5f2c5af..9ab97b288 100644
--- a/ui/build/kati.go
+++ b/ui/build/kati.go
@@ -85,7 +85,7 @@ func runKati(ctx Context, config Config, extraSuffix string, args []string, envF
 		// using 'include'.
 		"--no_ninja_prelude",
 		// Support declaring phony outputs in AOSP Ninja.
-		"--use_ninja_phony_output",
+//		"--use_ninja_phony_output",
 		// Support declaring symlink outputs in AOSP Ninja.
 		"--use_ninja_symlink_outputs",
 		// Regenerate the Ninja file if environment inputs have changed. e.g.
diff --git a/ui/build/ninja.go b/ui/build/ninja.go
index 5961c4525..e3b59481e 100644
--- a/ui/build/ninja.go
+++ b/ui/build/ninja.go
@@ -65,7 +65,7 @@ func runNinjaForBuild(ctx Context, config Config) {
 	args = append(args, "-f", config.CombinedNinjaFile())
 
 	args = append(args,
-		"-o", "usesphonyoutputs=yes",
+//		"-o", "usesphonyoutputs=yes",
 		"-w", "dupbuild=err",
 		"-w", "missingdepfile=err")
 
diff --git a/ui/build/soong.go b/ui/build/soong.go
index 87818e386..51e359400 100644
--- a/ui/build/soong.go
+++ b/ui/build/soong.go
@@ -272,11 +272,11 @@ func runSoong(ctx Context, config Config) {
 			config.PrebuiltBuildTool("ninja"),
 			"-d", "keepdepfile",
 			"-d", "stats",
-			"-o", "usesphonyoutputs=yes",
-			"-o", "preremoveoutputs=yes",
+//			"-o", "usesphonyoutputs=yes",
+//			"-o", "preremoveoutputs=yes",
 			"-w", "dupbuild=err",
-			"-w", "outputdir=err",
-			"-w", "missingoutfile=err",
+//			"-w", "outputdir=err",
+//			"-w", "missingoutfile=err",
 			"-j", strconv.Itoa(config.Parallel()),
 			"--frontend_file", fifo,
 			"-f", filepath.Join(config.SoongOutDir(), file))
-- 
2.17.1

